docs/csharp/whats-new/csharp-version-history.md

```diff
@@ -0,0 +1,124 @@
+- [Iterators](https://msdn.microsoft.com/en-us/library/65zzykke(v=vs.100).aspx)
+- [Covariance and contravariance](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/covariance-contravariance/)
+
+While C# may have started as a pretty generic object-oriented language, C# Version 2 changed that in a hurry.  Once they had their feet under them, they went after some serious developer pain points.  And they went after them in a big way.
+
+With generics, you have types and methods that can operate on an arbitrary type while still retaining type safety.  So, for instance, having a List&lt;T&gt; lets you have List&lt;string&gt; or List&lt;int&gt;  and perform type safe operations on those strings or ints while you iterate through them.  This certainly beats creating ListInt inheritors or casting from Object for every operation.
+
+Oh, and speaking of iterators, C# Version 2 brought iterators.  To put it succinctly, this let you iterate through the items in List (or other Enumerable types) with a foreach loop.  Having this as a first class part of the language dramatically enhanced readability of the language and people&#39;s ability to reason about the code.
+
+And yet, C# continued to play a bit of catch up with Java.  Java had already released versions that included generics and iterators.  But that would soon change as the languages continued to evolve apart.
+
+## C# Version 3
+
+C# Version 3 came in late 2007, along with Visual Studio 2008, though the full boat of language features would actually come with C# Version 3.5.  And what a version this proved to be.  This established C# as a truly formidable programming language.  Let&#39;s take a look at some major features in this version.
+
+- [Auto implemented properties](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/auto-implemented-properties)
+- [Anonymous types](http://www.c-sharpcorner.com/UploadFile/ff2f08/anonymous-types-in-C-Sharp/)
+- [Query expressions](https://docs.microsoft.com/en-us/dotnet/csharp/linq/query-expression-basics)
+- [Lambda expression](https://www.daedtech.com/introduction-to-c-lambda-expressions/)
+- [Expression trees](https://blogs.msdn.microsoft.com/charlie/2008/01/31/expression-tree-basics/)
+- [Extension methods](https://www.codeproject.com/Tips/709310/Extension-Method-In-Csharp)
+
+In retrospect, many of these features seem both inevitable and inseparable.  They all fit together so strategically.  It's generally thought that C# Version 3&#39;s killer feature was the query expression, also known as Linq ( **L** anguage **IN** tegrated **Q** uery).
+
+A more nuanced view examines expression tress, lamba expressions and anonymous types as the foundation upon which LINQ is constructed.  But, in either case, C# 3 presented with a fairly revolutionary concept.  C# 3 had begun to lay the groundwork for turning C# into a hybrid OO-functional language.
```

Comment:
typo: lamba

missing comma -> lambda expressions, and anonymous types

to use an acronym, you have to introduce it at first similar to the example I used before with LINQ